presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -289,22 +288,22 @@ public QueryInfo createQuery(Session session, String query)
             QueryExecution execution = new FailedQueryExecution(queryId, query, session, self, transactionManager, queryExecutor, e);
 
             queries.put(queryId, execution);
-            queryMonitor.createdEvent(execution.getQueryInfo());
-            queryMonitor.completionEvent(execution.getQueryInfo());
-            stats.queryFinished(execution.getQueryInfo());
+            QueryInfo queryInfo = execution.getQueryInfo();
+            queryMonitor.createdEvent(queryInfo);
+            queryMonitor.completionEvent(queryInfo);
+            stats.queryFinished(queryInfo);
             expirationQueue.add(execution);
 
-            return execution.getQueryInfo();
+            return queryInfo;
         }
 
-        queryMonitor.createdEvent(queryExecution.getQueryInfo());
+        QueryInfo queryInfo = queryExecution.getQueryInfo();
+        queryMonitor.createdEvent(queryInfo);
 
         queryExecution.addStateChangeListener(newValue -> {
             if (newValue.isDone()) {
-                QueryInfo info = queryExecution.getQueryInfo();
-
-                stats.queryFinished(info);
-                queryMonitor.completionEvent(info);
+                stats.queryFinished(queryInfo);
+                queryMonitor.completionEvent(queryInfo);
```

Comment:
I don't think you want to make this change.... this code is in a callback lambda